src/vcs/git.rs

```diff
@@ -788,6 +789,55 @@ impl<'a> Browser<'a> {
         self.repository.revision_branches(&commit.id())
     }
 
+    /// Get the [`Stats`] of the underlying [`Repository`].
+    ///
+    /// # Errors
+    ///
+    /// * [`error::Error::Git`]
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use radicle_surf::vcs::git::{Browser, Repository};
+    /// # use std::error::Error;
+    ///
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// let repo = Repository::new("./data/git-platinum")?;
+    /// let mut browser = Browser::new(&repo)?;
+    ///
+    /// let stats = browser.get_stats()?;
+    ///
+    /// // there should be 5 branches (local & remote)
+    /// assert_eq!(stats.branch_count, 5);
+    ///
+    /// // there should be 14 commits
+    /// assert_eq!(stats.commit_count, 14);
+    ///
+    /// // 4 contributors: Alexander Simmerl, Fintan Halpenny, FintanH, Rudolfs Ošiņš
+    /// assert_eq!(stats.contributor_count, 4);
+    ///
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn get_stats(&self) -> Result<Stats, Error> {
+        let history = self.repository.head()?;
+
+        let branch_count = self.list_branches(None)?.len();
+        let commit_count = history.iter().count();
+        let mut contributors: Vec<String> =
+            history.iter().cloned().map(|x| x.author.name).collect();
```

Comment:
Something that I think we need to think about is that this will change if the `Browser` switches branches. For example, if `master` is the default branch then we have all the contributors that have commits on that branch. But say there's another branch that is behind or ahead of master, then theoretically it will have less or more contributors.

Reply:
All good :ok_hand: https://github.com/radicle-dev/radicle-surf/issues/124#issuecomment-639332052